### DISCUSSION TITLE
help on challenge error

### DISCUSSION THREAD

**Discussion by ssine-med** [Cyfrin/foundry-full-course-cu#734] [Q&A] [answered]
hi, i'm a beginner and i don't know if i'm following the right way to solve the challenges,
all i do is copy the contract code into Remix trying to compile and deploy it, but i keep getting the same error in both first and second lessons :
 Error: not found AFoundryCourseChallenge.sol

**Comment by Niferu** [comment]
Hi @ssine-med 

You are doing it wrong then. Once you find challenge contract, visit it on etherscan then open code tab and search for comment `CALL THIS FUNCTION!`. You will find small hint there on how to solve challenge -> you need to do something to call `solvechallenge` function correct way, once you do that it will reward you.

Note that those challenges are real challanges (not so easy to solve) good luck :)

**Reply by ssine-med** [reply]
First of all, thank you @Neftyr for your time.
  I still don't know the correct steps to try out my ideas to solve the challenge. For example, where should I write the code? Should it be on Etherscan (where I can't modify it), or in the Blockscan IDE? Additionally, after I modify the code in Blockscan, I'm unsure how to verify if my solution is correct or not.
 All of these aspects are new to me, unlike to you :-) 👍

**Reply by Niferu** [reply, marked as answer]
Hi @ssine-med 

You want to basically call function named `solvechallenge`, which is coded in contract you have found by QR code. You already know how to call functions from external contracts right? You can call it directly on etherscan using metamask, you can also call it writing script in foundry and of course you can call it using your own contract using foundry or remix it's up to you.

If you still do not know how to call external contract functions keep progressing with course. Those challenges are pretty hard even for experienced developers, so do not waste time trying to solve them until you get enough knowledge and understanding. Just follow with course if you did not finished it and if you did you should definitely redo all lessons with extra attention to details and making sure you understand eveything. Once you do that get back to challenges ;)

**Reply by ssine-med** [reply]
Now it makes sense. I've only finished the first 2 lessons of the course. I thought that I missed something in those 2 lessons, which caused me to not be able to solve the challenge. I don't know how to thank you @Neftyr 🙏; you've helped me get through a huge disappointment. Now I'm going to leave the challenges until I finish the entire course. Once again, thank you.

**Reply by theirrationalone** [reply]
@Neftyr !, Incredible directions 🔥🔥
